Updating `@babel/core` to version 7.22.15+ shakes out needed exports

Running the following code results in a `TypeError: (0 , _$b_2.fn3) is not a function` error.

index.js
```js
import { fn1 } from './a';

fn1();
```

a.js
```js
import { fn3 } from './b';

export const fn2 = () => {
    fn3();
};

export const fn1 = () => {
    fn2();
};
```

b.js
```js
export const fn3 = () => {};
```

It could be related to the way exports are transpiled, it changed from this:
```js
exports.var = var;
var fn = function fn(arg) {
```

To this:
```js
var fn = exports.fn = function fn(arg) {
```

<details open>
  <summary>@babel/core 7.22.11</summary>

```js
(function(){function r(e,n,t){function o(i,f){if(!n[i]){if(!e[i]){var c="function"==typeof require&&require;if(!f&&c)return c(i,!0);if(u)return u(i,!0);var a=new Error("Cannot find module '"+i+"'");throw a.code="MODULE_NOT_FOUND",a}var p=n[i]={exports:{}};e[i][0].call(p.exports,function(r){var n=e[i][1][r];return o(n||r)},p,p.exports,r,e,n,t)}return n[i].exports}for(var u="function"==typeof require&&require,i=0;i<t.length;i++)o(t[i]);return o}return r})()({1:[function(require,module,exports){
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
/* common-shake removed: exports.fn2 = */ exports.fn1 = void 0;
var _b = require("./b");
var fn2 = function fn2() {
  (0, _b.fn3)();
};
/* common-shake removed: exports.fn2 = */ void fn2;
var fn1 = function fn1() {
  fn2();
};
exports.fn1 = fn1;

},{"./b":2}],2:[function(require,module,exports){
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.fn3 = void 0;
var fn3 = function fn3() {};
exports.fn3 = fn3;

},{}],3:[function(require,module,exports){
"use strict";

var _a = require("./a");
(0, _a.fn1)();

},{"./a":1}]},{},[3])
```
</details>

<details open>
  <summary>@babel/core 7.22.15</summary>

```js
(function(){function r(e,n,t){function o(i,f){if(!n[i]){if(!e[i]){var c="function"==typeof require&&require;if(!f&&c)return c(i,!0);if(u)return u(i,!0);var a=new Error("Cannot find module '"+i+"'");throw a.code="MODULE_NOT_FOUND",a}var p=n[i]={exports:{}};e[i][0].call(p.exports,function(r){var n=e[i][1][r];return o(n||r)},p,p.exports,r,e,n,t)}return n[i].exports}for(var u="function"==typeof require&&require,i=0;i<t.length;i++)o(t[i]);return o}return r})()({1:[function(require,module,exports){
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
/* common-shake removed: exports.fn2 = */ exports.fn1 = void 0;
var _b = require("./b");
var fn2 = /* common-shake removed: exports.fn2 = */ function fn2() {
  (0, _b.fn3)();
};
var fn1 = exports.fn1 = function fn1() {
  fn2();
};

},{"./b":2}],2:[function(require,module,exports){
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
/* common-shake removed: exports.fn3 = */ void void 0;
var fn3 = /* common-shake removed: exports.fn3 = */ function fn3() {};

},{}],3:[function(require,module,exports){
"use strict";

var _a = require("./a");
(0, _a.fn1)();

},{"./a":1}]},{},[3])
```
</details>

--- a/lib/shake/analyzer.js
+++ b/lib/shake/analyzer.js
@@ -427,6 +427,9 @@ Analyzer.prototype.siftMember = function siftMember(node, current, recursive,
   const dependsParent = ancestors && ancestors.find(parent => {
     if (parent.type !== 'AssignmentExpression') return false;
     if (parent.left.type !== 'MemberExpression') return false;
+    if (parent.left.type !== 'AssignmentExpression' && parent.right.id) {
+      return false;
+    }
     return this.isExports(parent.left.object);
   });
   const dependsNode = dependsParent && dependsParent.left.property;

--- a/test/analyzer-test.js
+++ b/test/analyzer-test.js
@@ -975,4 +975,86 @@ describe('Analyzer', () => {
       declarations: [ 'dont' ]
     });
   });
+
+  it('should work with @babel/core 7.22.11', () => {
+    analyzer.run(parse(`
+      const { fn1 } = require('./a');
+      fn1();
+    `), 'root');
+
+    analyzer.run(parse(`
+      const { fn3 } = require("./b");
+      const fn2 = function fn2() {
+        fn3();
+      };
+      exports.fn2 = fn2;
+      const fn1 = function fn1() {
+        fn2();
+      };
+      exports.fn1 = fn1;
+    `), 'a');
+
+    analyzer.run(parse(`
+      exports.fn3 = void 0;
+      const fn3 = function fn3() {};
+      exports.fn3 = fn3;
+    `), 'b');
+
+    analyzer.resolve('root', './a', 'a');
+    analyzer.resolve('a', './b', 'b');
+
+    assert.deepEqual(analyzer.getModule('root').getInfo(), EMPTY);
+    assert.deepEqual(analyzer.getModule('a').getInfo(), {
+      removeImport: false,
+      bailouts: false,
+      uses: [ 'fn1' ],
+      declarations: [ 'fn2', 'fn1' ]
+    });
+    assert.deepEqual(analyzer.getModule('b').getInfo(), {
+      removeImport: false,
+      bailouts: false,
+      uses: ['fn3' ] ,
+      declarations: ['fn3', 'fn3']
+    });
+  });
+
+  it('should work with @babel/core 7.22.15+', () => {
+    analyzer.run(parse(`
+      const { fn1 } = require('./a');
+      fn1();
+    `), 'root');
+
+    analyzer.run(parse(`
+      exports.fn2 = exports.fn1 = void 0;
+      const b = require('./b');
+      const fn2 = exports.fn2 = function fn2() {
+        b.fn3();
+      };
+      const fn1 = exports.fn1 = function fn1() {
+        fn2();
+      };
+    `), 'a');
+
+    analyzer.run(parse(`
+      exports.fn3 = void 0;
+      const fn3 = exports.fn3 = function fn3() {};
+    `), 'b');
+
+    analyzer.resolve('root', './a', 'a');
+    analyzer.resolve('a', './b', 'b');
+
+    assert.deepEqual(analyzer.getModule('root').getInfo(), EMPTY);
+    assert.deepEqual(analyzer.getModule('a').getInfo(), {
+      removeImport: false,
+      bailouts: false,
+      uses: [ 'fn1' ],
+      declarations: [ 'fn2', 'fn1', 'fn2', 'fn1' ]
+    });
+    assert.deepEqual(analyzer.getModule('b').getInfo(), {
+      removeImport: false,
+      bailouts: false,
+      uses: [ 'fn3' ] ,
+      declarations: [ 'fn3', 'fn3' ]
+    });
+  });
 });

